### DISCUSSION TITLE
How to register a server route manually? (non file-based)

### DISCUSSION THREAD

**Discussion by hymair** [nuxt/nuxt#27433] [Questions]
As the title says, I cannot find any relevant info in the docs or google. I am implementing a CQRS-like architecture and want something like this:
```
~/server/modules/some-module
    /commands
        /some-command
            some-command.controller.ts
            some-command.command.ts
        /other-command
            ....
```
The route definition will be defined inside `some-command.controller.ts`.

**Comment by undermoonn** [comment]
https://nuxt.com/docs/guide/directory-structure/server#nested-router

Maybe export the route definition from `some-command.controller.ts` and import it in a all matched server api can be a good way :)

**Reply by hymair** [reply]
Ideally I would want the router instance exposed to the import context. Then we can import that and manually register routes.

```
import { router } from '#somewhere'

router.get(...)
```
